packages/indexer-agent/src/agent.ts

```diff
@@ -778,6 +788,10 @@ class Agent {
       ? BigNumber.from(rule.allocationAmount)
       : this.indexer.defaultAllocationAmount
     const desiredNumberOfAllocations = 1
+    // makes sense to me for allocation-specific rule to trumps others
+    const desiredAllocationMaxLifetime = rule?.allocationMaxLifetime
+      ? BigNumber.from(Math.max(1, rule.allocationMaxLifetime - 1))
+      : maxAllocationEpochs
```

Comment:
similar to the other comment that it makes sense to me for allocation-specific rule to trumps others